### ISSUE TITLE
Bug using 2 gpus

### ISSUE THREAD

**Issue by chrisrn** [tensorflow/models#2818]
### System information
- **What is the top-level directory of the model you are using**:
tensorflow/models/research/slim
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.4.0
- **Bazel version (if compiling from source)**:
- **CUDA/cuDNN version**: 8/6
- **GPU model and memory**: 1)GeForce GTX Titan, 2) GeForce GTX 1080
- **Exact command to reproduce**:
`python train_image_classifier.py \
  --train_dir=${TRAIN_DIR} \
  --dataset_name=DATASET_NAME \
  --dataset_split_name=train \
  --dataset_dir=${DATASET_DIR} \
  --clone_on_cpu=False \
  --num_clones=2 \
  --model_name=resnet_v2_101 \
  --max_number_of_steps=154526 \
  --batch_size=64 \
  --learning_rate=0.0001 \
  --learning_rate_decay_factor=0.1 \
  --end_learning_rate=0.00001 \
  --optimizer=momentum \
  --momentum=0.9 \
  --weight_decay=0.0001 \
  --train_image_size=180 \
  --save_interval_secs=60 \
  --save_summaries_secs=60 \
  --log_every_n_steps=100 \
  --checkpoint_path=${PRETRAINED_CHECKPOINT_DIR}/resnet_v2_101.ckpt \
  --checkpoint_exclude_scopes=resnet_v2_101/logits`

### Describe the problem
I am training resnet_v2_101 model on my custom data set. I composed the data set from the script for getting TFRecord shards, so the data are right and shuffled. The training process is really slow using 2 gpus and using the command **watch nvidia-smi** the gpus are not being utilized all the time. They can be at 80-100% both but suddenly they fall to 0% for a small amount of time. I think this is something that slows down the training. I spent a lot of time to check if the problem has to do with the reading process of the files but I cannot figure out the source.

**Comment by aselle** [comment]
@sguada, do you have any thoughts or suggestions? Is the 2 gpu mainly just to get the problem to fit into RAM, and it is expected the utilization is low? @chrisrn, you might also try the tensorflow garden version of resnet, but I believe it only supports 1 gpu right no @nealwu, please look as well.

**Comment by aselle** [comment]
Two more things suggested by @tfboyd,
1. Could you also provide how many imagfes/second you are getting as a training rate.
2. You are using two different types of GPUs. It is unlikely to get good load balancing when different gpu architectures are involved, you will likely have a bottl3eneck on one that makes the other wait longer.

**Comment by tfboyd** [comment]
A few things that might help get more info:

- What is your images/sec or global batch for 1 GPU and then 2 GPUs
- what is your batch-size?  I did not look at the code I assume you have the default but nice to know.

he SLIM examples use slightly older versions of the input pipeline but you should still be able to get decent scaling for 2 GPUs.

**Comment by chrisrn** [comment]
I confirmed the problem by running the benchmark code that tensorflow provides. Using batch-size=64 I got approximately 190 images/sec in 1 gpu and 240 images/sec in 2 gpus without being utilized all the time. So the problem still exists without using slim. And I believe that there is no problem with the gpus I use because training in 1 gpu only I have good utilization and also fast time.

**Comment by nealwu** [comment]
For the official folder version ('garden' version), that's correct, it currently just supports one GPU.

**Comment by chrisrn** [comment]
Now I have 2 gpus GTX 1080 which have 6.1 cuda compute capability. The speed issue seems to be resolved according to tensorflow benchmark code. The images/sec are now almost 2 times more. But the utilization issue still exists according to slim code. The gpus are not fully utilized during training and sometimes they are close to 0%. So I don't know if it is a tensorflow bug.

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.

**Comment by chrisrn** [comment]
@tensorflowbutler  I am not using tf slim now so I cannot reproduce the problem. So you can temporarily close this issue!
